scipy/spatial/transform/_rigidtransformation.pyx

```diff
@@ -0,0 +1,1899 @@
+        """Initialize from a rotation and translation.
+
+        When composing a rotation and translation, the rotation is applied
+        before the translation, such that
+        ``t = T.from_rottrans(rotation, translation)``
+        is equivalent to
+        ``t = T.from_rotation(rotation) * T.from_translation(translation)``.
+
+        When applying a transformation to a vector, the result is the same as
+        if the transformation was applied to the vector in the following way:
+        ``t.apply(vector) == translation + rotation.apply(vector)``
+
+        Parameters
+        ----------
+        rotation : `Rotation` instance
+            A single rotation or a stack of rotations.
+        translation : array_like, shape (N, 3) or (3,)
+            A single translation vector or a stack of translation vectors.
+
+        Returns
+        -------
+        `RigidTransformation`, either a single transformation or a stack of
+        transformations.
+            - If rotation is single and translation is shape (3,), then a
+              single transformation is returned.
+            - Otherwise, a stack of transformations is returned.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as T
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Creating from a single rotation and translation:
+
+        >>> r = R.from_euler("ZYX", [90, 30, 0], degrees=True)
+        >>> r.as_matrix()
+        array([[ 0.       , -1.,  0.        ],
+               [ 0.8660254,  0.,  0.5       ],
+               [-0.5      ,  0.,  0.8660254 ]])
+        >>> d = np.array([2, 3, 4])
+        >>> t = T.from_rottrans(r, d)
+        >>> t.rotation.as_matrix()
+        array([[ 0.       , -1.,  0.        ],
+               [ 0.8660254,  0.,  0.5       ],
+               [-0.5      ,  0.,  0.8660254 ]])
+        >>> t.translation
+        array([2., 3., 4.])
+        >>> t.single
+        True
+
+        When applying a transformation to a vector, the result is the same as
+        if the transformation was applied to the vector in the following way:
+        ``t.apply(vector) == translation + rotation.apply(vector)``
+
+        >>> r.apply([1, 0, 0])
+        array([0.       , 0.8660254, -0.5     ])
+        >>> d + r.apply([1, 0, 0])
+        array([2.       , 3.8660254,  3.5     ])
+        >>> t.apply([1, 0, 0])
+        array([2.       , 3.8660254,  3.5     ])
+        """
+        return cls.from_translation(translation) * cls.from_rotation(rotation)
+
+    @cython.embedsignature(True)
+    @classmethod
+    def from_expcoords(cls, expcoords):
+        r"""Initialize from exponential coordinates of transformation.
+
+        This implements the exponential map that converts 6-dimensional real
+        vectors to SE(3). The first three components of the vector encode
+        the rotation and the last three components encode the translation.
```

Comment:
Can it be expressed as 4x4 matrix exponential `T = exp(S)`? Where `S` has the form
```
[skew(v) t]
[0       1]
```
With `v` being the rotation vector, `t` being the translation vector and `skew` is the skew matrix corresponding to a vector?

If so, perhaps this technical detail can be but in the docstring too (properly formatted)

Reply:
Let's say the exponential coordinates are $\mathcal{S}\theta$ with the first three components $\omega \theta$ and the last three components $v \theta$. We can put the same 6 numbers in such a 4x4 matrix as you pointed out. I would call it $\left[ \mathcal{S} \right] \theta$. There is an exponential map $T = \exp([\mathcal{S}] \theta)$. Sometimes $Exp$ is used for the vector and $\exp$ for the matrix to distinguish both exponential maps. Or we can just define this function as $T = \exp([\mathcal{S}] \theta)$.

Reply:
With $\theta$ being the rotation (and translation in other formulas?) vector norm?

Reply:
$\mathcal{S}$ is the screw axis and $\theta$ is the angle for pure rotations, yes. In case of combined rotation and translations I would call it the "magnitude of the displacement". Not sure if there is a better name for it. It is not the norm of the 6d vector. Either $v$ (for pure translations) or $\omega$ (other cases) have unit norm, but when $\omega$ has unit norm, $v$ might still be nonzero.